### PR TITLE
nixos/gdm: use systemd tmpfiles instead of preStart

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
Thet was a mess before. This should be more reliable.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Review by hedning — APPROVED** [review]
Ah, yes, much nicer :+1:

**Comment by worldofpeace** [comment]
For ref the syntax https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html.

File: nixos/modules/services/x11/display-managers/gdm.nix [hunk 1]
```diff
@@ -159,22 +159,21 @@ in
...
+        preStart = optionalString (defaultSessionName != null) ''
           # Set default session in session chooser to a specified values – basically ignore session history.
           ${setSessionScript}/bin/set-session ${cfg.sessionData.autologinSession}
         '';
       };
 
+    systemd.tmpfiles.rules = [
+      "d /run/gdm/.config 0711 gdm gdm -"
+    ] ++ optionals config.hardware.pulseaudio.enable [
+      "L+ /run/gdm/.config/pulse - - - - ${pulseConfig}"
```
Comment (jtojnar) [inline nixos/modules/services/x11/display-managers/gdm.nix]: Could we add a comment why is this needed?

Comment (worldofpeace) [reply nixos/modules/services/x11/display-managers/gdm.nix]: Possibly, but from this diff it looks undercomented. pulseConfig is prominently commented above, I'm not sure this was commented before.

**Comment by worldofpeace** [comment]
@GrahamcOfBorg test gnome3

File: nixos/modules/services/x11/display-managers/gdm.nix [hunk 2]
```diff
@@ -159,22 +159,21 @@ in
...
+    ] ++ optionals config.services.gnome3.gnome-initial-setup.enable [
+      # Create stamp file for gnome-initial-setup to prevent it starting in GDM.
+      "f /run/gdm/.config/gnome-initial-setup-done 0711 gdm gdm yes"
```
Comment (andersk) [inline nixos/modules/services/x11/display-managers/gdm.nix]: This causes `nixos-rebuild switch` to fail:

```
setting up tmpfiles
[/etc/tmpfiles.d/nixos.conf:7] Invalid age 'yes'.
warning: error(s) occurred while switching to the new configuration
```
